ui/src/app/cohort-search/demographics/demographics.component.tsx

```diff
@@ -0,0 +1,592 @@
+      } else {
+        this.setState({count: null});
+      }
+    });
+  }
+
+  updateAgeSelection() {
+    const {ageType, maxAge, minAge} = this.state;
+    const selectedNode = {
+      ...ageNode,
+      name: `Age In Range ${minAge} - ${maxAge}`,
+      attributes: [{
+        name: ageType,
+        operator: Operator.BETWEEN,
+        operands: [minAge, maxAge]
+      }],
+    };
+    this.props.select(selectedNode);
+  }
+
+  centerAgeCount() {
+    if (serverConfigStore.getValue().enableCBAgeTypeOptions && !!this.slider) {
+      // get range from slider element and convert the strings to numbers
+      const [sliderMin, sliderMax] = this.slider.get().map(v => +v);
+      // get width as a % by dividing the selected age range by the full slider range
+      const width = (sliderMax - sliderMin) / (+defaultMaxAge - +defaultMinAge) * 100;
+      // get left margin as a % by dividing the change in minAge by the full slider range
+      const marginLeft = (sliderMin - +defaultMinAge) / (+defaultMaxAge - +defaultMinAge) * 100;
+      const wrapper = document.getElementById('count-wrapper');
+      if (!!wrapper) {
+        wrapper.setAttribute('style', `margin-left: ${marginLeft}%; width: ${width}%; text-align: center;`);
+        // set style properties also for cross-browser compatibility
+        wrapper.style.marginLeft = `${marginLeft}%`;
+        wrapper.style.width = `${width}%`;
+        wrapper.style.textAlign = 'center';
+      }
+    }
+  }
+
+  selectOption(opt: any) {
+    triggerEvent('Cohort Builder Search', 'Click', `Demo - ${typeToTitle(opt.type)} - ${opt.name}`);
+    this.props.select({...opt, name: `${typeToTitle(opt.type)} - ${opt.name}`});
+  }
+
+  calculate(init?: boolean) {
+    let count = 0;
+    this.props.selections.forEach(sp => {
+      if (init) {
+        const node = this.state.nodes.find(n => n.conceptId === sp.conceptId);
+        if (node) {
+          sp.count = node.count;
+        }
+      }
+      count += sp.count;
+    });
+    this.setState({count});
+  }
+
+  calculateAge(init?: boolean) {
+    const {maxAge, minAge} = this.state;
+    if (!init || this.setTotalAge) {
+      this.setState({calculating: true});
+    }
+    const {cdrVersionId} = currentWorkspaceStore.getValue();
+    const min = init ? defaultMinAge : minAge;
+    const max = init ? defaultMaxAge : maxAge;
+    const parameter = {
+      ...ageNode,
+      name: `Age In Range ${min} - ${max}`,
+      attributes: [{
+        name: AttrName.AGE,
+        operator: Operator.BETWEEN,
+        operands: [min, max]
+      }],
+    };
+    const request = {
+      excludes: [],
+      includes: [{
+        items: [{
+          type: DomainType.PERSON.toString(),
+          searchParameters: [mapParameter(parameter)],
+          modifiers: []
+        }],
+        temporal: false
+      }],
+      dataFilters: []
+    };
+    cohortBuilderApi().countParticipants(+cdrVersionId, request).then(response => {
```

Comment:
Same as above, we want to try to prefer await statements rather than `.then`.

Reply:
Done.